### ISSUE TITLE
Lost all my cookies (browser sessions) after Brave crash

### ISSUE THREAD

**Issue by mayeaux** [brave/brave-browser#4492]
This is the second time that this is happened to me, Brave freezes and crashes and then when I fire it up back up I have lost all my sessions and have to relogin everywhere I use a cookie to maintain a session. This is really, really tiresome and even has implications to me as a professional developer, losing all my sessions could cause me a lot of trouble.

I looked for other issues but couldn't find any pertaining to this, I either have to prevent this or get a new browser, what can I do to prevent it?

**Comment by rebron** [comment]
Hi @mayeaux. We need some more information. Can you provide your Brave version and OS, it'll be in brave://version. Also, can you take a look at brave://crashes and send the crash reports. If you have the crash id, please post here. 

Is there a restore tabs option when you relaunch the browser?

**Comment by srirambv** [comment]
I encountered this on my dev profile as well. After restart the active tab loaded, when I tried to switch to a background tab, the tab froze and then instantly browser crashed. History doesn't have restore previous windows/tabs. There was no restore message shown when I relaunched the window. 

Happened on 

Brave | 0.66.66 Chromium: 74.0.3729.131 (Official Build) dev(64-bit)
-- | --
Revision | 518a41c1fa7ce1c8bb5e22346e82e42b4d76a96f-refs/branch-heads/3729@{#954}
OS | Linux

**Comment by srirambv** [comment]
Just encountreded the issue again on Nightly profile. Release profile was active. Updated nightly channel via `sudo apt update && sudo apt upgrade` on Linux. Launched nightly profile after upgrade and instantly crashed. No restore option showed up after relaunch and all tabs were lost. Found this crash id(1a40e5f06ba025ce) in brave://crashes but unable to open the crash report. Had set clear browsing data on exit so can't even see history now. 

Brave | 0.67.62 Chromium: 75.0.3770.38 (Official Build)nightly (64-bit)
-- | --
Revision | 3860105745f2b12537da9e9f048f14c3f52ba970-refs/branch-heads/3770@{#618}
OS | Linux

**Comment by ThomDap** [comment]
This just happened to me (lost all cookies after crash). This is terrible loose of time. I'm considering switching to another browser.
Version 0.66.99 Chromium: 75.0.3770.100
macOS 10.14.5 (18F132)

**Comment by rebron** [comment]
@ThomDap Hi sorry to hear this happened. Wondering if you could help us diagnose this so it doesn't happen again, we're having trouble reproducing. 

A couple things: 1) did the restore tabs option dialog show after the crash? 2) Would you be able to paste your crash ID under brave://crashes and also send the crash report. 3) Do you recall how many tabs you had opened?

**Comment by ThomDap** [comment]
1) Yes the restore tabs option dialog did show up but as usual I prefer to re-open manually tabs I want
2) The crash hadn't been recorded in brave://crashes and I have this written : "Crash reporting is disabled. Crash reporting is not available in Brave.". Don't know if this has got something to do with the absence of the crash report.
3) I would say a dozen.

I remember a macOS dialog opening just before the crash asking for my password to access the keychain (with the option Deny / Allow / Always allow) - and then crash - and then logget out of dozens of websites with 2FA etc.........

Nice to have a quick reply, hope I am of some help.

**Comment by rebron** [comment]
@ThomDap Yes very helpful, thanks for the additional info. We're still needing to nail down the full steps to reproduce and will dig further to see how others are running into this issue.

**Comment by tdy2014** [comment]
> Hi @mayeaux. We need some more information. Can you provide your Brave version and OS, it'll be in brave://version. Also, can you take a look at brave://crashes and send the crash reports. If you have the crash id, please post here.
> 
> Is there a restore tabs option when you relaunch the browser?

There has been ZERO change in this condition since initial report. Problem exists STILL in 

Brave | 0.67.124 Chromium: 76.0.3809.100 (Official Build) (64-bit)
-- | --
Revision | ed9d447d30203dc5069e540f05079e493fc1c132-refs/branch-heads/3809@{#990}
OS | Windows 10 OS Version 1903 (Build 18362.267)
JavaScript | V8 7.6.303.29
Flash | (Disabled)
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36
Command Line | "C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" --enable-dom-distiller --disable-domain-reliability --disable-chrome-google-url-tracking-client --no-pings --enable-features=NewExtensionUpdaterService --disable-features=AutofillServerCommunication,AudioServiceOutOfProcess,DefaultEnableOopRasterization,NetworkService,UnifiedConsent,LookalikeUrlNavigationSuggestionsUI,TranslateUI --flag-switches-begin --flag-switches-end
Executable Path | C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\????\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Variations | 5040e8a-5040e8ac134752e-f685ead32c707b42-dcf7018a411b6d4e-3f4a17dfd01ab0d3-60c33fc016e0dd70-3f4a17df414bf2d2-3f4a17df4ca682fe-3f4a17df551157b-3f4a17dff88a4127-2874767e74994832-3f4a17dfb7e2524c-c92d551c411c711e-3f4a17df8f029c91-3f4a17df1734fcac-3f4a17df61832c80-3f4a17dfcc20827f-77c422c2db4845-ace4e13874785582-3f4a17df4fd665d-cf4f6ead41e765a5-3f4a17df2a1c4112-3f4a17df776de70c-e0278d3ded9149a9-3f4a17dfa5cb8590-3f4a17df6c6c4f61-3f4a17df76b48ab8-d4bd105cc70841c8-a25670072a565533-11b3d144d8e87453-3f4a17dfd8343fa3-3f4a17dfb517d646-3f4a17df8ee5ed19-c3125ea274658432-fa79b94c8e667a85-3f4a17dfc36cdb78-3f4a17df591576c8-3f4a17df5274eb09-3f4a17dfd6d4ee9a-3f4a17df2729b628-3f4a17df44827ee5-3f4a17df8ca44045-3f4a17df3f79ac13-3d47f4f42ea1f2c6-3f4a17dff742e6d6-3f4a17dfde47491b-3f4a17df1beda558-1beda5589b4c4257-3f4a17df8fa604e0-ace4e13843f62d3b-3f4a17df9e5c75f1-30e1b12b2594bdf4-3f4a17df6fa07eb4-7282ca762981bcb4-3f4a17df5139837c-3f4a17df4934552d-3f4a17df7f8176d9-3f4a17df9504402-d0e5ea2a73a8d7f1-1b1a7908f7217a71-b204717823a898eb-fc93cf747a5ba892-3f4a17dfd1cd70a5-3f4a17df4ea303a6-3f4a17dfac8d4708-52dc3a48b19465ab-b19465abe5373d71-e23a25d96e6e0c7e-3f17a7d895876445-2b71ffe42b33233e-881ca6c94da5ae82-3f4a17df56302f8c-2f882e708d1fbf5c-3f4a17dfb2612322-8a9180b2cac0a91c-3f4a17dfad6d27cc-85d38e8dcec9a952-9a6c5085b8d7a428-76bcaaa6a372e609-3f4a17dff3ea30a0-3f4a17df2e7f6029-88947c67d840bbac-3f4a17dffb57f531-3f4a17df930e30f5-3f4a17df732f0134-3f4a17df9a2f4e5b-9ded16b91354da85-f1a864dc6844d8aa-669a04e0494d8760-3f4a17dff9e5da91-508355f5b1ceb06f-6b7b34dcc122d59-3f4a17df3a4029d-d7f231f351bb4372-3f4a17dfb0ea13bc-3f4a17df47a2af3f-3f4a17df2eb45dac-3f4a17df94e68624-803f8fc4cc73f8a1-3f4a17df7e91b7bd-c6da37cb6204e469-3f4a17dfbb61e032-bb61e0328d0eb7e3-3f4a17df41ddf744-c78ce173

As of now this browser is useless annoying garbage. Uninstalling. I helped develop Netscape. I'll build my own browser. tyvm

**Comment by bsclifton** [comment]
We have solved many crash problems in 0.68.x and above (currently, BETA channel). @tdy2014 if you wanted to try beta, you can grab a copy at https://brave.com/download-beta/

**Comment by cdobrich** [comment]
I just grabbed the latest stable version of Brave and the problem is still happening on Linux. Doesn't matter if there is a crash or if I exit the program cleanly. Regularly I lose all my cookies and have to log into everything again. @bsclifton

**Comment by ThomDap** [comment]
Just happened once again. **Absolutely unbearable**. Changing browser for good right away.

brave://version

Brave	0.70.121 Chromium: 78.0.3904.70 (Official Build) (64-bit)
Revision	edb9c9f3de0247fd912a77b7f6cae7447f6d3ad5-refs/branch-heads/3904@{#800}
OS	macOS Version 10.14.6 (Build 18G95)
JavaScript	V8 7.8.279.17
Flash	(Disabled)
User Agent	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.70 Safari/537.36
Command Line	/Applications/Brave Browser.app/Contents/MacOS/Brave Browser --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=NewExtensionUpdaterService,PasswordImport,SimplifyHttpsIndicator,WebUIDarkMode --disable-features=AutofillServerCommunication,AudioServiceOutOfProcess,SmsReceiver,LookalikeUrlNavigationSuggestionsUI,NotificationTriggers,UnifiedConsent,SyncUSSBookmarks --flag-switches-begin --disable-features=AutofillServerCommunication,AudioServiceOutOfProcess,SmsReceiver,LookalikeUrlNavigationSuggestionsUI,NotificationTriggers,UnifiedConsent,SyncUSSBookmarks,DisallowUnsafeHttpDownloads --flag-switches-end
Executable Path	/Applications/Brave Browser.app/Contents/MacOS/Brave Browser
Profile Path	/Users/Thomas/Library/Application Support/BraveSoftware/Brave-Browser/Default

**Comment by chopin2712** [comment]
This also happened for the second time for me. I am on Linux and my computer has crash (I don't know why but my session has closed) and when I opened it again the sound was lost. I solved my sound issue quickly but for the Brave cookies, I lost them again. 

This is my Brave version:

Brave	0.70.123 Chromium: 78.0.3904.97 (Official Build) (64-bit)
Revision	021b9028c246d820be17a10e5b393ee90f41375e-refs/branch-heads/3904@{#859}
OS	Linux
JavaScript	V8 7.8.279.23
Flash	(Disabled)
User Agent	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.97 Safari/537.36
Command Line	/opt/brave.com/brave/brave --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=NewExtensionUpdaterService,PasswordImport,SimplifyHttpsIndicator,WebUIDarkMode --disable-features=LookalikeUrlNavigationSuggestionsUI,AutofillServerCommunication,NotificationTriggers,SmsReceiver,AudioServiceOutOfProcess,UnifiedConsent,SyncUSSBookmarks --flag-switches-begin --flag-switches-end
Executable Path	/opt/brave.com/brave/brave
Profile Path	/home/chopin/.config/BraveSoftware/Brave-Browser/Default

I cannot access to brave://crashes.

The problem hasn't been solved.

**Comment by mayeaux** [comment]
Yeah I've personally just been using Chrome since this happened to me the second time. Unacceptable for this to happen.

**Comment by ThomDap** [comment]
> Yeah I've personally just been using Chrome since this happened to me the second time. Unacceptable for this to happen.

Same here, just went back to old Chrome - flawless since then.

**Comment by oshanz** [comment]
what is happening here. I lost all sessions and saved passwords are missing. so frustrating incident 

> Version 0.70.121 Chromium: 78.0.3904.70 (Official Build) (64-bit)
> Linux 5.0.0-31-generic #33~18.04.1-Ubuntu SMP Tue Oct 1 10:20:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by mayeaux** [comment]
Guys just use Chrome until they are able to fix this. Not worth it otherwise sadly.

**Comment by bsclifton** [comment]
I definitely understand how frustrating this can be; having to re-login to all your sites is super annoying. I've lost cookies before on Chrome too. Unfortunately, troubleshooting can be difficult. The times I lost my cookies, the root cause was differences in the sqlite schema.  For example, if I switched from Canary to Release channel. Usually, there are DB migrations on new Chromium releases which are executed and handle upgrading the DB format (which fails if you go backwards a Chromium version)

Specifically, the cookies are stored in a sqlite DB in your profile directory. For me on macOS, this is at `~/Library/Application Support/BraveSoftware/Brave-Browser-Nightly/Default/Cookies`. Whenever this gets corrupted (ex: when we upgrade from Chromium 78 to Chromium 79; when a cookie related crash happens) I believe the DB is erased, which is why cookies are lost.

Assuming folks here are not switching between Brave versions (Release / Beta / Developer / Nightly)... so I'm thinking there must be a crash is happening. Would be great to know if something is shown under brave://crashes for folks affected

brave://crashes should definitely be accessible. @chopin2712 do you have info about why you can't access it? I don't think it's block-able by group policy, but just in case you're using Brave at work, you can check via brave://policy to see if your computer falls into that

When brave://crashes is shown, if there are crashes there, they may not be uploaded (by default, we turn crash reporting off for release channel). If that is the case, there will be a button for sending. You can click that and then restart the browser and you should be able to verify it's sent. Specifically, we need the ID from `Uploaded Crash Report ID <ID HERE>` (not the `Local Context` one)

**Comment by cdobrich** [comment]
I can confirm this is still happening. It seems like only part of the cookies are lost sometimes and the rest of the time it's all of them. Sigh....

**Comment by mayeaux** [comment]
@bsclifton I've personally never lost cookies with Chrome but I had it happen twice to me within about a month with Brave. It's sad because I like and support Brave but I can't use broken software and this to me is too broken to use.

**Comment by bsclifton** [comment]
@mayeaux does anything show under brave://crashes? If you visit that page, it'll show recent crash reports and let you send them to us (which helps us identify the root cause)

We would absolutely like to solve this problem but we're in a situation where we need information to help diagnose where the problem is. @mayeaux what OS are you on? and you were trying with recent versions? Which release channel? (stable?)

With help, we should be able to track down specific problems and fix them 😄

**Comment by oshanz** [comment]
brave version: Version 1.1.23 Chromium: 79.0.3945.88 (Official Build) (64-bit) 
there are no crashes at the brave://crashes. I just close the window. at the next load I got all the tabs but without any session. I had to login again :-[]

**Comment by bsclifton** [comment]
I'm going to close this issue as there is nothing actionable we can do at the moment. There haven't been any responses with crash IDs we can investigate and there has been no activity here for the past 3 months.  

If you're encountering this issue from a search, please create a new issue and visit brave://crashes to find recent crashes. You can send the report- you may need to close/re-open browser to have the report actually sent (it happens in background, not sure how to give it higher priority to send). Once sent, there will be a `Report ID` field which is important to share with us. We can see where the problem is and work towards finding the root cause

**Comment by mayeaux** [comment]
Has happened again. I won't ever use Brave again, a shame!

**Comment by cdobrich** [comment]
> Has happened again. I won't ever use Brave again, a shame!

I had this problem A LOT, then I did some digging and investigation on my system. I uncovered TWO problems that seemed to be the cause of it. (1) For some stupid reason, my system had my Brave browser version several versions behind. I don't know how this happened since I love checking for software updates every morning. But when I manually forced it to update, it started working again. (2) I deleted my old browser profile and created a fresh new one. This was a good thing since my old one had a lot of cruft, with the folder being 6 GB.
